In [375]:
import pandas as pd
%matplotlib inline

In [376]:
df = pd.read_csv('tmdb-movies.csv')

To get a list of columns

In [377]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10866 entries, 0 to 10865
Data columns (total 21 columns):
id                      10866 non-null int64
imdb_id                 10856 non-null object
popularity              10866 non-null float64
budget                  10866 non-null int64
revenue                 10866 non-null int64
original_title          10866 non-null object
cast                    10790 non-null object
homepage                2936 non-null object
director                10822 non-null object
tagline                 8042 non-null object
keywords                9373 non-null object
overview                10862 non-null object
runtime                 10866 non-null int64
genres                  10843 non-null object
production_companies    9836 non-null object
release_date            10866 non-null object
vote_count              10866 non-null int64
vote_average            10866 non-null float64
release_year            10866 non-null int64
budget_adj              1

Just before proceeding with any thing, let's drop all dumpicates

In [378]:
duplicate_records = sum(df.duplicated())
df.drop_duplicates(inplace=True)
print('{} duplicate records have been deleted'.format(duplicate_records))

1 duplicate records have been deleted


From the `df.info()` does show that there is a number of missing values, to show how many are missing:

In [379]:
df.isnull().sum().sort_values(ascending=False)

homepage                7929
tagline                 2824
keywords                1493
production_companies    1030
cast                      76
director                  44
genres                    23
imdb_id                   10
overview                   4
popularity                 0
budget                     0
revenue                    0
original_title             0
revenue_adj                0
budget_adj                 0
runtime                    0
release_date               0
vote_count                 0
vote_average               0
release_year               0
id                         0
dtype: int64

Some of these columns are not needed, no need to waste any resource processing them: such as imdb_id, homepage

In [380]:
df.drop(labels=['imdb_id', 'homepage', 'tagline', 'overview', 'keywords', 'production_companies', 'id'], axis=1, inplace=True)

In [381]:
df_clean = df.dropna(axis=0)
df.shape[0] - df_clean.shape[0]

134

In [382]:
# To get any abnormalities:
df_clean.describe()

popularity        budget       revenue       runtime    vote_count  \
count  10731.000000  1.073100e+04  1.073100e+04  10731.000000  10731.000000   
mean       0.652615  1.480365e+07  4.031989e+07    102.468829    219.812972   
std        1.004804  3.106456e+07  1.176524e+08     30.493873    578.815324   
min        0.000188  0.000000e+00  0.000000e+00      0.000000     10.000000   
25%        0.210765  0.000000e+00  0.000000e+00     90.000000     17.000000   
50%        0.387081  0.000000e+00  0.000000e+00     99.000000     39.000000   
75%        0.720889  1.600000e+07  2.500000e+07    112.000000    148.000000   
max       32.985763  4.250000e+08  2.781506e+09    900.000000   9767.000000   

       vote_average  release_year    budget_adj   revenue_adj  
count  10731.000000  10731.000000  1.073100e+04  1.073100e+04  
mean       5.964710   2001.259622  1.776530e+07  5.200623e+07  
std        0.930283     12.820151  3.446630e+07  1.454252e+08  
min        1.500000   1960.000000  0.000000e+00  0.000000e+00  
25%        5.400000   1995.000000  0.000000e+00  0.000000e+00  
50%        6.000000   2006.000000  0.000000e+00  0.000000e+00  
75%        6.600000   2011.000000  2.110885e+07  3.470546e+07  
max        9.200000   2015.000000  4.250000e+08  2.827124e+09

Let's see how many unique cast we have:

In [383]:
s = set()
for cast in df_clean['cast']:
    s.update(cast.split('|'))
print("There are: {num_casts} unique cast".format(num_casts=len(s)))

There are: 18930 unique cast


One initial idea I had is to add 18930 column to the dataframe, the index name of those are the cast names, and values are boolean which represent whether the cast has been in the movie or not. But let's be honest, such details are useless. Another approach, let's see how many casts listed per each movie:

In [384]:
s = set()
for cast in df_clean['cast']:
    s.update([len(cast.split('|'))])
print(s)

{1, 2, 3, 4, 5}


So, the number of casts listed for each movie is between 1 and 5.

In [386]:
df_clean.head()

popularity     budget     revenue                original_title  \
0   32.985763  150000000  1513528810                Jurassic World   
1   28.419936  150000000   378436354            Mad Max: Fury Road   
2   13.112507  110000000   295238201                     Insurgent   
3   11.173104  200000000  2068178225  Star Wars: The Force Awakens   
4    9.335014  190000000  1506249360                     Furious 7   

                                                cast          director  \
0  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   Colin Trevorrow   
1  Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...     George Miller   
2  Shailene Woodley|Theo James|Kate Winslet|Ansel...  Robert Schwentke   
3  Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...       J.J. Abrams   
4  Vin Diesel|Paul Walker|Jason Statham|Michelle ...         James Wan   

   runtime                                     genres release_date  \
0      124  Action|Adventure|Science Fiction|Thriller       6/9/15   
1      120  Action|Adventure|Science Fiction|Thriller      5/13/15   
2      119         Adventure|Science Fiction|Thriller      3/18/15   
3      136   Action|Adventure|Science Fiction|Fantasy     12/15/15   
4      137                      Action|Crime|Thriller       4/1/15   

   vote_count  vote_average  release_year    budget_adj   revenue_adj  
0        5562           6.5          2015  1.379999e+08  1.392446e+09  
1        6185           7.1          2015  1.379999e+08  3.481613e+08  
2        2480           6.3          2015  1.012000e+08  2.716190e+08  
3        5292           7.5          2015  1.839999e+08  1.902723e+09  
4        2947           7.3          2015  1.747999e+08  1.385749e+09

In [368]:
def extract_actor(t, i):
    print(t, i)
    try:
        return t.split('|')[i]
    except IndexError:
        return ''
    
df_clean = df_clean.assign(actor1=extract_actor(df_clean.cast, 0))

0        Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...
1        Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...
2        Shailene Woodley|Theo James|Kate Winslet|Ansel...
3        Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...
4        Vin Diesel|Paul Walker|Jason Statham|Michelle ...
5        Leonardo DiCaprio|Tom Hardy|Will Poulter|Domhn...
6        Arnold Schwarzenegger|Jason Clarke|Emilia Clar...
7        Matt Damon|Jessica Chastain|Kristen Wiig|Jeff ...
8        Sandra Bullock|Jon Hamm|Michael Keaton|Allison...
9        Amy Poehler|Phyllis Smith|Richard Kind|Bill Ha...
10       Daniel Craig|Christoph Waltz|Léa Seydoux|Ralp...
11       Mila Kunis|Channing Tatum|Sean Bean|Eddie Redm...
12       Domhnall Gleeson|Alicia Vikander|Oscar Isaac|S...
13       Adam Sandler|Michelle Monaghan|Peter Dinklage|...
14       Robert Downey Jr.|Chris Hemsworth|Mark Ruffalo...
15       Samuel L. Jackson|Kurt Russell|Jennifer Jason ...
16       Liam Neeson|Forest Whitaker|Maggie Grace|Famke.

AttributeError: 'Series' object has no attribute 'split'

In [374]:
df_clean['cast'].isna().any()

False